lib/diagnostic.rb

```diff
@@ -18,17 +18,19 @@ def example
 # In a Ruby comment, explain Behavior Driven Development, how it is meant to be
 # used, and how it differs from Test Driven Development.
 
-# your answer here
-
+Behavioral driven development begins with a User story and is based on the actions a user
+will perform. Test Driven development is focused more on satisfying tests that are created
+prior to coding.
 #
 # Question 2
 #
 # Create a request spec for our Examples API that ensures 'GET /examples'
 # responds successfully and lists all examples.
 
 RSpec.describe 'Examples API' do
-  # your test(s) here
-end
+  def examples
+  example.all
+  end
 
```

Comment:
You need quite a bit more here. And remember that every `do` needs to be paired with an `end`.

```rb
RSpec.describe 'Examples API' do
  def examples
    Example.all
  end

  def example
    Example.first
  end

  it 'lists all examples' do
    get '/examples'

    expect(response).to be_success

    examples_response = JSON.parse(response.body)
    expect(examples_response.length).to eq(examples.count)
    expect(examples_response['examples'].first['title']).to eq(example.title)
  end
end
```